WeatherApp/WeatherApp/App/DI/Dependencies.swift

```diff
@@ -0,0 +1,66 @@
+import UIKit
+
+protocol UseCaseDependenciesProtocol {
+
+    var getWeatherUseCase: GetWeatherUseCaseProtocol { get }
+
+}
```

Comment:
You can remove this protocol in favour of creating getWeatherUseCase private to this class

It would make sense if someone was using this protocol somewhere, but currently its only used as a list of all the use cases which does not really add additional benefits to the codebase